numba_dppy/tests/test_debug.py

```diff
@@ -0,0 +1,91 @@
+#! /usr/bin/env python
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import pytest
+
+import numba_dppy as dppy
+import dpctl
+
+from numba.core import types, compiler
+from numba_dppy import compiler
+from numba_dppy.tests.skip_tests import skip_test
+
+
+# TODO: Add level0 and move to common place
+offload_devices = [
+    "opencl:gpu:0",
+    "opencl:cpu:0",
+]
+
+
+@pytest.fixture(params=offload_devices)
+def offload_device(request):
+    return request.param
+
+
+def get_kernel_ir(fn, sig, debug=False):
+    kernel = compiler.compile_kernel(fn.sycl_queue, fn.py_func, sig, None, debug=debug)
+    return kernel.assembly
+
+
+def make_check(fn, sig, expect):
```

Comment:
This function makes a lot:
1. compile (with signature and debug options)
2. use regexp to search for debug symbols
3. make an assertion
To much responsibilities for the function. Also when it is called from test code it usually requires to see inside it to understand what exactly it does.

Reply:
It's just incapsulated to this function. Ok, I can move 1 and 3 to test